RFC-0001-row-id.md

```diff
@@ -0,0 +1,80 @@
+# **RFC 1 for Presto**
+
+## Row ID
+
+Proposers
+
+* Brian Dahmen
+* Devang Shah
+* Elliotte Rusty Harold
+* Serge Druzkin
+* Sreeni Viswanadha
+
+## Related Issues
+
+* Row ID support #22078
+* Binary IDs for Hive partitions #22103
+* Binary rowID type #22053
+* Start adding row\_id hidden column to Hive #22008
+
+
+## Summary
+
+A row ID is a binary key for a row that is globally unique across the warehouse.
+It identifies a specific row at a specific version of a partition’s data. A row
+ID will continue to locate the same row in the warehouse until the underlying
+partition is mutated.
+
+## Background
+
+Queries can produce lists of row IDs that can be saved and used as input to
+future queries for quick selection. A list of row IDs can be used include or
+exclude rows to form a subset from a table.
+
+### Goals
+
+Row IDs resolve to the same value across compute engines. In this way they can
+be used as a strong-equality comparison across compute engines.
+
+### Non-goals
+
+Comparison and sorting of row IDs.
+
+## Proposed Implementation
+
+A row ID is a VARBINARY pseudo-column, also known as a hidden column. Each row
+will have a unique value that can be compared for equality, byte by byte. The
+type is not orderable. In SQL queries this will have the name `$row_id` and the
+type ROW_ID, a new built-in type. The complete row ID will have two components,
+the first to identify a row within a partition and the second to identify the
+partition within the warehouse.
+
+The exact format of the row ID is connector specific and not constrained by this
+RFC. It might change from one warehouse to the next if different warehouses use
+different metatdata servers to supply the partition half of a row ID.
+
+The metadata server (e.g. Hive Metastore) supplies the partition component in
```

Comment:
Not everything uses a Hive metastore for its catalog.  Does this mean this proposal is specific to the Hive connector?

Reply:
No, that's why "e.g." instead of "i.e.".  I'll spell it out. Good tech writing doesn't use Latin. :-)